base/files/file_util.cc

```diff
@@ -369,7 +369,45 @@ bool ReadFileToStringWithMaxSize(const FilePath& path,
     return false;
   return ReadStreamToStringWithMaxSize(file_stream.get(), max_size, contents);
 #endif
+#if defined(STARBOARD)
```

Comment:
Let's make this part of the above macro so we know we'd like to clean it up

Also I know @sherryzy was looking at implementing the above types already so the upstream code worked as is, did anything come from that?

Reply:
I agree with that, but if Sherry has already gotten the upstream code working, we can just use that. If not, then let's use this